wagtail_newsletter/templates/wagtail_newsletter/panels/newsletter_panel.html

```diff
@@ -51,118 +47,167 @@
 
     {% else %}
 
-        {% if has_action_permission %}
-            <div class="help-block help-info">
-                {% icon name="help" %}
-                These actions will save a new page revision.
+        <p>
+            <div class="wn-panel--buttons">
+                {% if has_action_permission.save_campaign %}
+                    <button
+                        type="button"
+                        class="button button-secondary button-longrunning"
+                        data-a11y-dialog-show="wn-save-dialog"
+                        data-controller="w-progress"
+                        data-action="wn-submit:save_campaign@window->w-progress#activate"
+                    >
+                        {% icon name="spinner" %}
+                        Save campaign to {{ backend_name }}
+                    </button>
+                {% endif %}
+
+                {% if has_action_permission.send_test_email %}
+                    <button
+                        type="button"
+                        class="button button-secondary button-longrunning"
+                        data-a11y-dialog-show="wn-test-dialog"
+                        data-controller="w-progress"
+                        data-action="wn-submit:send_test_email@window->w-progress#activate"
+                    >
+                        {% icon name="spinner" %}
+                        Send test email
+                    </button>
+                {% endif %}
+
+                {% if has_action_permission.send_campaign %}
+                    <button
+                        type="button"
+                        class="button button-primary button-longrunning"
+                        data-controller="w-progress"
+                        data-wn-panel-target="sendButton"
+                        data-action="
+                            wn-panel#clickSend
+                            wn-submit:send_campaign@window->w-progress#activate
+                        "
+                    >
+                        {% icon name="spinner" %}
+                        Send campaign
+                    </button>
+                {% endif %}
             </div>
-        {% endif %}
 
-        <p>
-            <input
-                type="hidden"
-                name="newsletter-test-email"
-                data-wn-panel-target="testAddress"
-            >
+            {% dialog icon_name="mail" id="wn-save-dialog" title="Save campaign" dialog_root_selector="[data-edit-form]" %}
```

Comment:
The only reason for this dialog is to follow the pattern of the other two actions, so we get the benefit of having the real submit button be at the end of the form.